src/pubtools/_marketplacesvm/arguments.py

```diff
@@ -166,18 +166,43 @@ class RepoQueryLoad(Action):
     This action is intended to allow the optional load of mappings right in the command call
     instead of having to request data from server.
 
-    It will evaluate the input data and set it as a StArMap's QueryResponse.
+    It will evaluate the input data and set it as a StArMap's QueryResponseContainer.
     """
 
     def __init__(self, *args, **kwargs):
         """Instantiate the RepoQueryLoad action."""
         super(RepoQueryLoad, self).__init__(*args, **kwargs)
 
     def __call__(self, parser, namespace, values, options=None):
-        """Convert the received args into QueryResponse."""
+        """Convert the received args into QueryResponseContainer."""
         items = getattr(namespace, self.dest, None) or []
         if values and isinstance(values, str):
             items = json.loads(values)
             if not isinstance(items, list):
                 raise ArgumentError(self, f"Expected value to be a list, got: {type(items)}")
         setattr(namespace, self.dest, items)
+
+
+class RepoFileQueryLoad(Action):
+    """
+    Argparse Action subclass for loading StArMap mappings from the ``repo-file`` argument.
+
+    This action is intended to allow the optional load of mappings from a json file
+    instead of having to request data from server.
+
+    It will evaluate the input file and set it as a StArMap's QueryResponseContainer.
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Instantiate the RepoQueryLoad action."""
+        super(RepoFileQueryLoad, self).__init__(*args, **kwargs)
+
+    def __call__(self, parser, namespace, values, options=None):
+        """Convert the received args into QueryResponseContainer."""
+        items = getattr(namespace, self.dest, None) or []
+        if values and isinstance(values, str):
+            with open(values, "r") as v:
+                items = json.load(v)
```

Comment:
@jajreidy @JAVGan you were too fast guys :-)

Could we use yaml as an input format?
Each Json si syntactically valid YAML format
and staging structure is in yaml :-)

BTW one can convert json from starmap response to yaml quite easily ;-)
```
curl --fail -s https://starmap.url/api/v2/query?image=rhel-image-9.4-1 | yq -o yaml -P
```
sorry for post merge request